snappi_ixnetwork/trafficitem.py

```diff
@@ -191,9 +220,245 @@ class TrafficItem(CustomField):
+            if "ipv4" in string:
+                tr_type = "ipv4"
+            elif "ipv6" in string:
+                tr_type = "ipv6"
+            else:
+                tr_type = "ethernetVlan"
+            paths[d]["type"] = tr_type
+        return paths
+
+    def get_ixn_config(self, config):
+        ixn = self._api.assistant._ixnetwork
+        myfilter = [{"property": "name", "regex": ".*"}]
+        url, payload = self._get_search_payload(
+            "/traffic",
+            "(?i)^(trafficItem|configElement|frameRate"
+            "|frameSize|transmissionControl|stack|field|highLevelStream"
+            "|tracking|transmissionDistribution)$",
+            [
+                "name",
+                "trafficType",
+                "type",
+                "rate",
+                "duration",
+                "displayName",
+                "valueFormat",
+            ],
+            myfilter,
+        )
+        self.ixn_config = None
+        tr = self.create_traffic(config)
+        imports = {}
+        imports["traffic"] = tr
+        self._importconfig(imports)
+        return ixn._connection._execute(url, payload)
+
+    def remove_ixn_traffic(self):
+        if len(self._api._ixnetwork.Traffic.TrafficItem.find()) > 0:
+            with Timer(self._api, "Remove Flows"):
+                start_states = [
+                    "txStopWatchExpected",
+                    "locked",
+                    "started",
+                    "startedWaitingForStats",
+                    "startedWaitingForStreams",
+                    "stoppedWaitingForStats",
+                ]
+                state = self._api._ixnetwork.Traffic.State
+                if state in start_states:
+                    self._api._ixnetwork.Traffic.StopStatelessTrafficBlocking()
+                url = "%s/traffic/trafficItem" % self._api._ixnetwork.href
+                self._api._request("DELETE", url)
+                self._api._ixnetwork.Traffic.TrafficItem.find().refresh()
+        self.traffic_index = 1
+
+    def create_traffic(self, config):
+        flows = config.flows
+        tr = {"xpath": "/traffic", "trafficItem": []}
+        ports = self.get_ports_xpath(config)
+        devices = self.get_device_xpath(config)
+        for i, f in enumerate(flows):
+            if f._properties.get("name") is None:
+                raise Exception("name shall not be null for flows")
+            if f._properties.get("tx_rx") is None:
+                msg = (
+                    "Please configure the flow endpoint"
+                    "for flow indexed at %s" % i
+                )
+                raise Exception(msg)
+            if f.tx_rx.choice is None:
+                msg = "Flow endpoint needs to be either port or device"
+                raise Exception(msg)
+            if f.tx_rx.choice == "port":
+                tr_type = "raw"
+                ep = getattr(f.tx_rx, "port")
+                tx_objs = ["%s/protocols" % ports.get(ep.tx_name)]
```

Comment:
"SrcDestMesh" section is missing which is handle by self._get_mesh_type(flow)

Reply:
yes mesh handling is not added, because this implementation was way older when we are defaulting the mesh type to manytomany. will add mesh handling